quint/src/effects/inferrer.ts

```diff
@@ -87,7 +87,7 @@ export class EffectInferrer implements IRVisitor {
   private freeNames: { effectVariables: Set<string>; entityVariables: Set<string> }[] = []
 
   // the current depth of operator definitions: top-level defs are depth 0
-  private definitionDepth: number = 0
```

Comment:
I am not sure about what's going on with the variable `definitionDepth` here.  Is it getting decremented anywhere? Also, it looks like the comment is not in sync with the initializer.

Reply:
It is manipulated by the `walk*` functions, which is how the visitors are supposed to be used. The comment is correct, if you look at `this.definitionDepth` at some point and its value is 0, then you are in a top level definition.

I don't like this initialization here, but I couldn't think of a better way to do it. I wonder if extending this from a class (on top of implementing the interface) is better or worse.